modules/core/src/test/java/org/apache/ignite/internal/processors/cache/AbstractDataTypesCoverageTest.java

```diff
@@ -534,4 +540,200 @@ public void nestedObjectField(ObjectBasedOnPrimitivesAndCollections nestedObjFie
+        }
+
+        /**
+         * Constructor.
+         *
+         * @param time Original local time value.
+         */
+        public Timed(LocalTime time) {
+            val = time;
+            sqlStrVal = "PARSEDATETIME('" + TIME_DATE_FORMAT.format(
+                java.sql.Time.valueOf(time)) + "', '" + PATTERN + "')";
+        }
+
+        /** @inheritDoc */
+        @Override public Object originalVal() {
+            return val;
+        }
+
+        /** @inheritDoc */
+        @Override public String sqlStrVal() {
+            return sqlStrVal;
+        }
+    }
+
+    /**
+     * Holder for Date and Timestamp values.
+     */
+    public static class Dated implements SqlStrConvertedValHolder {
+        /** */
+        private static final long serialVersionUID = 0L;
+
+        /** */
+        private static final String PATTERN = "yyyy-MM-dd HH:mm:ss.SSS";
+
+        /** */
+        private static final SimpleDateFormat DATE_TIME_DATE_FORMAT = new SimpleDateFormat(PATTERN);
+
+        /** Original value. */
+        private Object val;
+
+        /** Converted value. */
+        private String sqlStrVal;
+
+        /**
+         * Constructor.
+         *
+         * @param date Original value.
+         */
+        public Dated(Date date) {
+            val = date;
+            sqlStrVal = "PARSEDATETIME('" + DATE_TIME_DATE_FORMAT.format(date) + "', '" + PATTERN + "')";
+        }
+
+        /**
+         * Constructor.
+         *
+         * @param date Original LocalDateTime value.
+         */
+        public Dated(LocalDateTime date) {
+            val = date;
+            sqlStrVal = "PARSEDATETIME('" + DATE_TIME_DATE_FORMAT.format(
+                java.sql.Timestamp.valueOf(date)) + "', '" + PATTERN + "')";
+        }
+
+        /**
+         * Constructor.
+         *
+         * @param date Original LocalDate value.
+         */
+        public Dated(LocalDate date) {
+            val = date;
+            sqlStrVal = "PARSEDATETIME('" + DATE_TIME_DATE_FORMAT.format(
+                java.sql.Timestamp.valueOf(date.atStartOfDay())) + "', '" + PATTERN + "')";
+        }
+
+        /** @inheritDoc */
+        @Override public Object originalVal() {
+            return val;
+        }
+
+        /** @inheritDoc */
+        @Override public String sqlStrVal() {
+            return sqlStrVal;
+        }
+    }
+
+    /**
+     * Holder for byte array values.
+     */
+    public static class ByteArrayed implements SqlStrConvertedValHolder {
+        /** */
+        private static final long serialVersionUID = 0L;
+
+        /** Original value. */
+        private Object val;
+
+        /** Converted value. */
+        private String sqlStrVal;
+
+        /**
+         * Constructor.
+         *
+         * @param byteArr Original value.
+         */
+        public ByteArrayed(byte[] byteArr) {
+            val = byteArr;
+            sqlStrVal = "x'" + Hex.encodeHexString(byteArr) + "'";
```

Comment:
There is no need to add a new dependency just for Hex function as
we already have U.byteArray2HexString()

Reply:
Fixed.